<launch>

  <node pkg="tf" type="static_transform_publisher" name="world" args="0 0 0 0 0 0 map world 100" /> 

	<node name="orb_slam2_mono" pkg="orb_slam2_ros" type="orb_slam2_ros_mono" output="screen">
        

      <remap from="/camera/image_raw" to="/tello/camera/image_raw" />

      <param name="publish_pointcloud" type="bool" value="true" />
      <param name="publish_pose" type="bool" value="true" />
      <param name="localize_only" type="bool" value="false" />
      <param name="reset_map" type="bool" value="false" />

      <!-- static parameters -->
      <param name="use_viewer" type="bool" value="true" />
      <param name="load_map" type="bool" value="false" />
      <param name="map_file" type="string" value="labaratory_rot.bin" />
      <param name="settings_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/config/Tello.yaml" />
      <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

      <param name="pointcloud_frame_id" type="string" value="map" />
      <param name="camera_frame_id" type="string" value="camera_link" />
      <param name="min_num_kf_in_map" type="int" value="15" />
  </node>

  <!-- <node name="image_muxer" pkg="flock_driver" type="image_muxer.py" output="screen" /> -->

  <node name="cloud_map_saver" pkg="flock_driver" type="cloud_map_saver.py" output="screen">
  <param name="OUT_FILE_PATH" type="string" value="/home/arkadiros/map_point.txt" />
  <param name="CLOUD_TOPIC_NAME" type="string" value="/orb_slam2_mono/map_points" />
  <param name="TRIGGER_TOPIC_NAME" type="string" value="/save_cloud_trigger" />
  <param name="POSE_TOPIC_NAME" type="string" value="/orb_slam2_mono/pose" />
  <param name="CAMERA_TOPIC_NAME" type="string" value="/orb_slam2_mono/debug_image" />

  
</node>


 <node name="flock_driver_node" pkg="flock_driver" type="flock_driver.py" output="screen" >
    <param name="network_interface" type="string" value="" />
    <param name="ID" type="string" value="" />

 

    <!-- <param name="network_interface" type="string" value="" /> -->
    <!-- <param name="TELLO_IP" type="string" value="192.168.43.98" /> -->

    <!-- <remap from="tello0/camera/image_raw" to="rectify/image_mono" /> -->
    <!-- <remap from="/rectify/image_mono" to="tello0/camera/image_mono" /> -->
    <!-- <remap from="tello0/camera/image_raw" to="/image" /> -->

 </node>


<node name="tello_slam_control" pkg="flock_driver" type="tello_slam_control.py" output="screen">
  <param name="~POSE_TOPIC_NAME" type="string" value="/orb_slam2_mono/pose" />
  <param name="ID" type="string" value="" />
</node>


<node name="tello_ui" pkg="flock_driver" type="tello_ui.py" output="screen">
  <param name="ID" type="string" value="" />
  <param name="~POSE_TOPIC_NAME" type="string" value="/orb_slam2_mono/pose" />
</node> 

  <!-- <node name="tello_keyboard_node" pkg="flock_driver" type="tello_keyboard.py" output="screen">
      <remap from="/tello/cmd_vel" to="/tello0/cmd_vel" />
      <remap from="/tello/takeoff" to="/tello0/takeoff" />
      <remap from="/tello/land" to="/tello0/land" />

  </node> -->



</launch>
